website/server/controllers/api-v3/tasks.js

```diff
@@ -838,6 +838,28 @@ api.moveTask = {
 
     // In memory updates
     const order = owner.tasksOrder[`${task.type}s`];
+    if (order.indexOf(task._id) === -1) { // task is missing from list, list needs repair
+      const taskListQuery = { type: task.type };
+      if (group) {
+        taskListQuery['group.id'] = owner._id;
```

Comment:
This group fix might also be beneficial in website/server/controllers/api-v3/tasks/groups.js although I'd guess that's a lower priority than the current issue, so perhaps just a TODO comment here?